webxr-api/session.rs

```diff
@@ -42,6 +42,42 @@ pub enum SessionMode {
     ImmersiveAR,
 }
 
+/// https://immersive-web.github.io/webxr/#dictdef-xrsessioninit
+#[derive(Clone, Debug, Eq, PartialEq)]
+#[cfg_attr(feature = "ipc", derive(Serialize, Deserialize))]
+pub struct SessionInit {
+    pub required_features: Vec<String>,
+    pub optional_features: Vec<String>,
+}
+
+impl SessionInit {
+    /// Helper function for validating a list of requested features against
+    /// a list of supported features for a given mode
+    pub fn validate(&self, mode: SessionMode, supported: &[String]) -> Result<Vec<String>, Error> {
```

Comment:
Is it worth converting these strings to atoms to avoid cloning them?

Reply:
They're going to be created once per session, that's not a big deal imo